src/v/raft/recovery_stm.cc

```diff
@@ -319,7 +314,13 @@ ss::future<> recovery_stm::send_install_snapshot_request() {
             .dirty_offset = _ptr->dirty_offset()};
 
           _sent_snapshot_bytes += chunk_size;
-
+          auto meta = get_follower_meta();
+          if (!meta) {
+              // stop recovery when node was removed
+              _stop_requested = true;
+              return ss::make_ready_future<>();
+          }
+          (*meta)->expected_log_end_offset = _ptr->_last_snapshot_index;
```

Comment:
On a second thought - maybe we shouldn't do it here for every install_snapshot request, but only for the last one? Otherwise the following sequence is possible:
1. only the snapshot on leader (dirty_offset == last_snapshot_index)
2. leader sends install_snapshot part1 to the follower
3. replicate request comes, the should_skip check passes, we send the append_entries request to the follower
4. but it is still not ready, we need to send install_snapshot part2, etc.

Snapshot parts are not that different from recovery batches. So maybe set it only for `done=true`?

Reply:
good point